lib/foreman_azure_rm/engine.rb

```diff
@@ -42,7 +42,7 @@ class Engine < ::Rails::Engine
       require 'azure_mgmt_subscriptions'
 
       # Add format validation for azure images
-      ::Image.validates :uuid, format: { with: /\A((marketplace|custom|gallery):\/\/)[^:]+(:[^:]+:[^:]+:[^:]+)?\z/,
+      ::Image.validates :uuid, uniqueness: { case_sensitive: false }, format: { with: /\A((marketplace|custom|gallery):\/\/)[^:]+(:[^:]+:[^:]+:[^:]+)?\z/,
```

Comment:
This means you cannot add the same image ID to different Azure compute resources.
You should use `scope` to limit the uniqueness. Example: https://github.com/theforeman/foreman/blob/develop/app/models/image.rb#L13

Reply:
Thanks @ShimShtein for the catch! Made changes.